cmd/deployment/extension/create_test.go

```diff
@@ -0,0 +1,151 @@
+
+	"github.com/elastic/ecctl/cmd/util/testutils"
+)
+
+func Test_createCmd(t *testing.T) {
+	createRawResp, err := ioutil.ReadFile("./testdata/create.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var succeedResp = new(models.Extension)
+	if err := succeedResp.UnmarshalBinary(createRawResp); err != nil {
+		t.Fatal(err)
+	}
+
+	createJSONOutput, err := json.MarshalIndent(succeedResp, "", "  ")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		name string
+		args testutils.Args
+		want testutils.Assertion
+	}{
+		{
+			name: "fails due to empty argument",
+			args: testutils.Args{
+				Cmd: createCmd,
+				Args: []string{
+					"create",
+				},
+				Cfg: testutils.MockCfg{Responses: []mock.Response{
+					mock.SampleInternalError(),
+				}},
+			},
+			want: testutils.Assertion{
+				Err: `requires at least 1 arg(s), only received 0`,
+			},
+		},
+		{
+			name: "fails due to missing flags",
+			args: testutils.Args{
+				Cmd: createCmd,
+				Args: []string{
+					"create", "mybundle",
+				},
+				Cfg: testutils.MockCfg{Responses: []mock.Response{
+					mock.SampleInternalError(),
+				}},
+			},
+			want: testutils.Assertion{
+				Err: `required flag(s) "type", "version" not set`,
+			},
+		},
+		{
+			name: "fails due to mismatching flags",
+			args: testutils.Args{
+				Cmd: createCmd,
+				Args: []string{
+					"create", "mybundle", "--version", "*", "--type", "bundle",
+					"--download-url", "example.com", "--file", "hi.zip",
+				},
+				Cfg: testutils.MockCfg{Responses: []mock.Response{
+					mock.SampleInternalError(),
+				}},
+			},
+			want: testutils.Assertion{
+				Err: "both --file and --download-url are set. Only one may be used",
+			},
+		},
+		{
+			name: "fails due to API error",
+			args: testutils.Args{
+				Cmd: createCmd,
+				Args: []string{
+					"create", "mybundle", "--version", "*", "--type", "bundle",
+				},
+				Cfg: testutils.MockCfg{Responses: []mock.Response{
+					mock.SampleInternalError(),
+				}},
+			},
+			want: testutils.Assertion{
+				Err: mock.MultierrorInternalError.Error(),
+			},
+		},
+		{
+			name: "succeeds",
+			args: testutils.Args{
+				Cmd: createCmd,
+				Args: []string{
+					"create", "mybundle", "--version", "*", "--type", "bundle", "--description", "Why hello there",
+				},
+				Cfg: testutils.MockCfg{
+					OutputFormat: "json",
+					Responses: []mock.Response{
+						mock.New200ResponseAssertion(
+							&mock.RequestAssertion{
+								Header: api.DefaultWriteMockHeaders,
+								Method: "POST",
+								Path:   "/api/v1/deployments/extensions",
+								Host:   api.DefaultMockHost,
+								Body:   mock.NewStringBody(`{"description":"Why hello there","extension_type":"bundle","name":"mybundle","version":"*"}` + "\n"),
+							},
+							mock.NewByteBody(createRawResp),
+						),
+					},
+				},
+			},
+			want: testutils.Assertion{
+				Err: string(createJSONOutput) + "\n",
+			},
+			// Not testing for file upload as the content type is multipart/form-data
```

Comment:
We should add the tests case for the file, we don't need to assert the request though, only the response.